File: Set1_1000/204_Count_Primes.java
class Solution {
    public int countPrimes(int n) {
        boolean [] prime = new boolean[n];
        int primeCount = 0;
        for (int i=2; i*i<prime.length; i++){
            if (!prime[i]){
                for (int j=2; j*i<prime.length; j++)
                    prime[i*j] = true;
            }
        }
        for (int i=2; i<prime.length; i++){
            if (!prime[i])
                primeCount ++;
        }
        return primeCount;
    }
}

File: Set1_1000/204_Count_primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n < 3:
            return 0
        dp = [0, 0] + [1] * (n - 2)
        for i in range(2, int(n ** 0.5) + 1):
            if dp[i]:
                dp[i ** 2:n:i] = [0] * len(dp[i ** 2:n:i])
        return sum(dp)
                
